src/python/turicreate/data_structures/sframe.py

```diff
@@ -2188,7 +2188,10 @@ def _imagecols_to_stringcols(self, num_rows=10):
                 if t in image_column_names:
                     printed_sf[t] = self[t].astype(str)
         return printed_sf.head(num_rows)
-
+    
+    def drop_duplicates(self, subset): 
+        if subset[0] in self.column_names(): 
+            return self.groupby(subset[0], {col: aggregate.SELECT_ONE(col) for col in self.column_names() if col != subset[0]}) 
```

Comment:
I believe this only works with duplicates in the first column.  I believe that this should be: 

```
self.groupby(subset, {col: aggregate.SELECT_ONE(col) for col in self.column_names()})
```